src/Cheats.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0 <0.9.0;
+
+import "./Storage.sol";
+import "./Vm.sol";
+
+// Wrappers around Cheats to avoid footguns
+contract Cheats {
+    using stdStorage for StdStorage;
+
+    StdStorage private stdstore;
+    Vm private constant vm_cheats = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
```

Comment:
this should be inherited

```suggestion
    Vm internal constant vm_cheats = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
```

Reply:
You mean inherited in `Test`?

The problem with marking it as `internal` is that there would be collisions with the third-party testing frameworks which need to also define `vm` such that that framework can used independently of `forge-std` (for example, [I define `vm` in `PRBTest`](https://github.com/paulrberg/prb-test/blob/d781e55cc8eaf923b3eb4edb9a2a808a93c39a6a/src/PRBTest.sol#L65)).

I suggest leaving it as `private`. Forge Std already redeclares the `vm` in `Test`.

Reply:
I think a good approach to enabling composability long-term may be for individual components (`Cheats`, etc) to provide only the functionality they are made for.
E.g. `Cheats` should expose std-cheats; not `vm` because it's a Forge thing and not a feature of `Cheats`. This will also prevent collisions when composing, and some devs might come up with interesting combinations we cannot foresee.

Additionally, every component should be a self-contained unit, with its own private `vm` and `stdstore` if it needs it, so it does not depend on another component. I don't think that would be too much hassle.

Specifically made end-user contracts, such as `Test` and `Script` can expose `vm` and `stdstore`.

Reply:
Very much agree with @ZeroEkkusu's resolution here.

> every component should be a self-contained unit, with its own private `vm` and `stdstore`

+1